.github/workflows/upstream-dev-ci.yaml

```diff
@@ -14,7 +14,9 @@ jobs:
   upstream-dev:
     name: upstream-dev
     runs-on: ubuntu-latest
-    if: github.repository == 'pydata/xarray'
+    if: |
+      contains(github.event.head_commit.message, 'test-upstream')
+      || (github.repository == 'pydata/xarray' && github.event_name != 'pull_request')
```

Comment:
While thinking more about this, I realized that there are instances in which we may want to run upstream-dev CI during development (for instance, to make sure that the added changes work). So, instead of disabling the upstream-dev CI on push and pull request events, this condition will allow the committer to run upstream dev CI by adding `test-upstream` in their git commit message. 

Any thoughts on this are welcomed

Reply:
Yes we would want this when fixing an update failure.

Can we have a skip-ci check too?

Reply:
when would we need to use `skip-ci`? Unless I'm missing something we should never try to skip the scheduled CI, and the upstream-dev PR CI is only run when manually requested (using the actions menu or by including `test-upstream` in the commit message).

Reply:
> when would we need to use skip-ci?

For doc-only commits.

But this point you make below

> (especially since squash-merging concatenates the first comment of a PR and all of its commit messages

is a really good point. Perhaps we should just ignore my `skip-ci` suggestion.

Reply:
> For doc-only commits

I see. Yes, that would be really useful, but maybe we should discuss that in #4730?

> (especially since squash-merging concatenates the first comment of a PR and all of its commit messages

we might work around that by making sure these commands can only be in the first (or first three) lines of the commit message

Reply:
unfortunately, extracting lines from the commit message seems to be fairly complicated because github does not include functions for string manipulation, and people seem to either not do squash merges or use a separate action.

Reply:
I think we should defer this for now then. Seems like a lot of work for minor benefit especially now that CI takes approx 20 mins instead of 40

Reply:
we would probably have to do that for the upstream-dev CI anyways